rules/php73/src/Rector/FuncCall/RemoveMissingCompactVariableRector.php

```diff
@@ -72,20 +72,53 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
+        $this->unsetUnusedArrayElements($node, $scope);
+        $this->unsetUnusedArguments($node, $scope);
+
+        if ($node->args === []) {
+            return new Array_();
+        }
+
+        return $node;
+    }
+
+    private function unsetUnusedArrayElements(Node $node, Scope $scope): void
+    {
         foreach ($node->args as $key => $arg) {
-            $argValue = $this->getValue($arg->value);
+            if (! $arg->value instanceof Array_) {
+                continue;
+            }
+
+            $array = &$arg->value;
+
+            foreach ($array->items as $arrayKey => $item) {
+                $value = $this->getValue($item->value);
+                if ($scope->hasVariableType($value)->yes()) {
+                    continue;
+                }
+
+                unset($array->items[$arrayKey]);
+            }
 
+            if ($arg->value->items === []) {
+                unset($node->args[$key]);
+            }
```

Comment:
What are there 3 lines doing?

Reply:
Remove the whole array argument if all the variables in this array are missing

Reply:
I see. So removes empty `compact()`?
Could you we add this information somehow to the code, so it's obvious without asking?

Reply:
This needs fix too